### ISSUE TITLE
Proficiencies: Knowledge of Material and Technique

### ISSUE THREAD

**Issue by lcy03406** [CleverRaven/Cataclysm-DDA#44385]
# Is your feature request related to a problem? Please describe.

Now we have proficiencies, leather working, plastic working, forging etc. Many proficiencies are about a material or a technique. They are many and not categorized. We can make it more clear.

# Describe the solution you'd like

Implement a category system of proficiencies, with proper UI.
Add 3 categories of proficiencies: Material, Technique and Other. 
Tools can be another category, I'm not sure.
Maybe link proficiencies to skills.
For combat skills, weapon types can be another proficiency category.

For example:
make a bronze spear: fabrication skill, bronze knowledge, general forging technique, weapon forging technique.
make leather boots: talor skill, lether knowledge, boot making technique.
make a stone knife: suvival skill, stone knowledge, knapping technique, smoothing technique.
make meat soup: cooking skill, meat knowledge, stewing technique.
make lard: cooking skill, meat knowledge, frying technique.
make a pizza: cooking skill, wheat knowledge, meat knowledge, baking technique.

# Describe alternatives you've considered

None.

# Additional context

None.

**Comment by Giagantic** [comment]
I disagree with adding redundant or unnecessary proficiencies, especially in the case of cooking. If cooking were to have proficiences in it shouldn't result in you making it more slowly, that honesty makes very little sense as a whole, instead proficencies should primarily play a role in the resulting product itself. We can all in theory make a steak in similar time periods regardless of being a chef or being a cook, but the quality and quantity of the product will not necessarily be the same based on experience. (I think this idea should be the primary point to proficiencies in all honesty, someone with tailoring 1 should produce a inferior product to someone with tailoring 10) Prior to gaining a proficiency you should instead produce a product with lower nutritional value, and/or in lower amounts, also there could be ways of tying proficiency to the creation of mass batches of a product. 
Things like "wheat knowledge" is really silly, this is coming from someone that is a IRL Caterer and son of two chefs. There is no such things as "wheat knowledge" just call it for what it is, Baking. 

Better ideas for proficiency:
- Knife work
- Cooking 
- Baking
- Cocktails 
- Sauces

That is it, there is no need for anything else, not like we need a plating proficiency nor do we need to add separate categories for each form of meat and so forth. Knife Work covers everything from prep work with the ingredients to minor butchering beyond the initial stuff you'd perform at your butchers rack (removing tiny fish bones, cutting away the silver skin on beef, etc). Cooking and Baking obvious play a role how they are composed and finished, Cocktails for drinks and Sauces for well... sauces. Honestly, you could probably remove the sauces category, it is an official position in classic restaurants but unnecessary for CDDA.

**Comment by lcy03406** [comment]
> I disagree with adding redundant or unnecessary proficiencies,

i don't want but we already have rope twisting proficiency, etc. I just propose a category method. 

> If cooking were to have proficiences in it shouldn't result in you making it more slowly, instead proficencies should primarily play a role in the resulting product itself. 

I agree. My PR does not contain this part only because #44377 has said this idea.

**Comment by Giagantic** [comment]
> 
> 
> > I disagree with adding redundant or unnecessary proficiencies,
> 
> i don't want but we already have rope twisting proficiency, etc.
> 
> > If cooking were to have proficiences in it shouldn't result in you making it more slowly, instead proficencies should primarily play a role in the resulting product itself.
> 
> Agree. My PR does not contain this part only because #44384 has said this idea.

Thing with Rope making is that it actually does take a lot of time to do, compared to someone that's be practicing it in real life for years, it would take a very long time. Cooking doesn't have the same restrictions in regards to time, anyone can cook, the product just will differ.

**Comment by lcy03406** [comment]
wheat knowledge is not baking. in tailor skill there are fabric knowledge, weaving, rope twisting.  in cook skill there are wheat knowledge, baking, noodle making.

**Comment by natsirt721** [comment]
Specifics aside, Gigantic does bring up a good point that some proficiencies will primarily affect the quality of the outcome whereas others will primarily affect the work time.  Like all things, reality relies somewhere between the two extremes - proficiencies will _in general_ help me make things faster and better, on a sliding scale between the two (as compared to a total novice) but rarely _just_ faster or _just_ better. Quality can manifest in a number of ways:
- For general fabrication of single items (weapons, vehicle parts, clothing) the resulting item may be damaged.
- For general fabrication of multiple items (ammunition, nails, etc.) fewer items may be produced.  
- For cooking, the result may have lower nutrition, lower morale score, or be reduced in quantity.
- For chemistry, the craft may totally fail (you forgot to stir your solution and it separated! RIP materials), although this is/was mostly covered by success chance for crafting. Not sure if the proficiencies system ties into success chance, but I would expect it to or at least propose that it should if it doesn't already.
I haven't played with proficiencies enough to have strong opinions yet, so I'll probably have a more refined idea in a week or so once I get back into cataclysm.

**Comment by I-am-Erk** [comment]
Pretty much all proficiencies fall under "technique". The materials that are used as proficiency names are used because they require a specific technique. For example, redsmithing is separate from blacksmithing because it requires less heat and more beating/coldworking. Elastics is a proficiency of tailoring because there are specific techniques necessary for making stretch fabric fit properly. Etc. There's never going to be a "wheat" proficiency, because that's called "baking", the technique generally used to work with flour products.

At some point, proficiency impact will probably become more granular. If we get quality levels, proficiencies could impact that, for example. In the meantime we have the ability to change crafting time and failure rate, and I'm trying to be pretty careful in how I use those. For example, not knowing the basics of how to do something has a much higher impact on your failure rate than knowing the basics but not being experienced with it.

**Comment by lcy03406** [comment]
> There's never going to be a "wheat" proficiency, because that's called "baking", the technique generally used to work with flour products.

Do you bake noodles? You never read my post carefully.

Noodles are made by cutting or squeezing or stretching, and prepared by boiling or frying. You Never Bake Noodles.

**Comment by lcy03406** [comment]
Material familiarity is a simulation that once a persion is proficent in a technique about that material, they will do better in some other techniques than average person.

You may call "metal material master" as "blacksmith", if you feel it is wierd to name proficiencies with materials.

If you spend days forging armor plates, you will be good at blacksmith. But that is not enough to forge a masterwork sword in reasonale time.
You are better than a novice average man but not better than a trained weaponsmith. 
Simulated in game, you have blacksmith and armorsmith proficiencies but do not have weaponsmith proficiency.

**Comment by lcy03406** [comment]
![image](https://user-images.githubusercontent.com/3512151/94520660-39cfcb00-025f-11eb-9862-6dd5c0e782a2.png)

Some buns and dumplings are fried or steamed, not baked.

**Comment by I-am-Erk** [comment]
You are very hung up on this. Yes I am aware that baking is not literally the only thing you can do with wheat. I said as much. However nearly every proficiency related to working with wheat is going to fall under baking, and those that don't will fall into other technique based proficiencies. Working with wheat isn't a special skill of its own. Having made a lot of sourdough does not mean your wheat familiarity is going to make you better at making pasta. Likewise, doing a lot of whittling doesn't make you better at carpentry.

**Comment by lcy03406** [comment]
Nearly every proficiency related to working with wheat is tightly related. But they are not going to fall under baking. You can call "wheat knowledge" as "doughing".

My feature request is a category system of proficiencies.

Current:
All proficiencies are scattered in a huge list, including "blacksmith" and your many tailor related proficiencies, and maybe my "baking/doughing".

Suggest:
Split the huge list to at least two. Call priciples of leather working, plastic working, blacksmithing, "baking" simply as Leather, Plastic, Metal, Wheat.

My goal is that proficiencies are categorized, and every list is not too long.

**Comment by lcy03406** [comment]
Another category system is to sort proficiencies by skill. there will be 10~  proficiencies in tailor skill and similar in other skills. That's OK.

But then a problem. fabrication skill's proficiency list will be longer then other skills, and composed by some groups that not much related to each other. Should medieval blacksmithing techniques be a skill?

**Comment by lcy03406** [comment]
Another way is to organize proficiencies into a tree structure, similar to talent trees in many other games.

**Comment by I-am-Erk** [comment]
Proficiencies are already organized into a tree structure, and "doughing" is not a thing. Further, as someone considered generally quite good at making pasta, bread, and pastry, I don't agree that there's very much connection between them at all despite their nearly identical ingredients. Getting a good noodle consistency has almost no relationship to making good sourdough, and is almost exactly the opposite of making good pastry. I chose wheat because it's a well known example that I know a lot about.

For example, a simplified version of the leatherworking tree:
![image](https://user-images.githubusercontent.com/45136638/94585540-01b39280-0235-11eb-9fb6-1e14395f17ff.png)

Proficiencies are agnostic of skill. There is no 'grouping proficiencies by skill'.

In general I'd recommend becoming familiar with the proficiency system as it currently exists and *then* making your recommendations.

**Comment by Fris0uman** [comment]
So what's the point of this issue? To add a menu where you can see proficiencies ordered into some sort of category? Why?
This is a very vague issue, what are you actually asking for?

**Comment by lcy03406** [comment]
> In general I'd recommend becoming familiar with the proficiency system as it currently exists and _then_ making your recommendations.

I am not good at English so I will just ignore your manner of speaking. 

I didn't find your tree graph in game. 

apologize in advance if it does exist in game, please help me and point it out. 

if the graph is in your mind and you are implementing it, it's good news and this issue can close up. 

Have not read much ui code, but if you are not coding now, i can try to help a bit.

**Comment by I-am-Erk** [comment]
You can't find a tree graph in game, no. It would be complex and hard to follow.

Is what you want just a UI change? Because you haven't made that clear at any point in this conversation.

**Comment by lcy03406** [comment]
> Is what you want just a UI change? Because you haven't made that clear at any point in this conversation.

Not only a UI change. But both of us agree on a  tree structure. It's a good start. I will think how to represent a tree in tile/ascii ui.

**Comment by I-am-Erk** [comment]
It's not going to be possible dude, this is a very cross-connected relationship. That branching diagram doesn't show how prof_articulation is also required for chitin working, and also for armoursmithing, which itself requires blacksmithing.

What is it that you actually want? A way to browse available proficiencies?

**Comment by lcy03406** [comment]
> What is it that you actually want?

A category system.

You see, there are 14 proficiencies in tailoring. If you do other skills/professions like this, there will be handrands proficiencies. You can't say althrough Cobbling, Millinery, Knitting, Speed Knitting, Principles of Leatherworking and Leatherworking are all different proficiencies, but all things involve elecric (for example) will be only one proficiency.

My suggestion is based on the above assumption. There will be 100~ proficiencies finally.

Another assumption is that unlike mutations, players will eventually learn almost all recipe-related proficiencies.

So there will be a long list of learned proficiencies.

I think a long list of 100~ proficiencies is bad. So I suggest there should be some category system.

If you insist on calling "wheat knowledge" as some "Principles of Baking", it is OK. But there should be some other kind of category system.

**Comment by I-am-Erk** [comment]
If what you're concerned about here is just UI listing, yes, at some point there will be some categorization.

**Comment by Zireael07** [comment]
For now, there's an open PR adding the capability to search for proficiencies: https://github.com/CleverRaven/Cataclysm-DDA/pull/44509

**Comment by lcy03406** [comment]
> If what you're concerned about here is just UI listing, yes, at some point there will be some categorization.

OK. I once thought the material proficiencies of a recipe can be deduct automatically and we need not to modify every recipe to add proficiencies . But you wish to not bind proficiencies to skills, then I don't know whether "copper knowledge in weaponsmith" and "copper knowledge in electronics" should be one or two proficiencies. So I decide not to think that way.
Expecting your categorization.
